lib/auth/trustedcluster.go

```diff
@@ -0,0 +1,94 @@
+package auth
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/gravitational/teleport"
+
+	"github.com/gravitational/trace"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type ValidateTrustedClusterResponse struct {
+	CAs [][]byte `json:"certificate_authorities"`
+}
+
+type ValidateTrustedClusterRequest struct {
+	Token string   `json:"token"`
+	CAs   [][]byte `json:"certificate_authorities"`
+}
+
+func (s *AuthServer) RequestValidateTrustedCluster(proxyAddress string, vreq ValidateTrustedClusterRequest) (*ValidateTrustedClusterResponse, error) {
+	requestBody, err := json.Marshal(vreq)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	responseBody, err := s.requestToProxy(proxyAddress, requestBody)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var vres ValidateTrustedClusterResponse
+	err = json.Unmarshal(responseBody, &vres)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &vres, nil
+}
+
+func (s *AuthServer) ValidateTrustedCluster(token string) error {
+	roles, err := s.ValidateToken(token)
+	if err != nil {
+		return trace.AccessDenied("invalid token")
+	}
+
+	if !roles.Include(teleport.RoleTrustedCluster) {
+		return trace.AccessDenied("role does not match")
+	}
+
+	if !s.checkTokenTTL(token) {
+		return trace.AccessDenied("expired token")
+	}
+
+	return nil
+}
+
+func (s *AuthServer) requestToProxy(host string, body []byte) ([]byte, error) {
+	url := fmt.Sprintf("https://%v/webapi/trustedclusters/validate", host)
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(body))
```

Comment:
can you please use `roundtrip` client library for that instead of implementing it again.